src/lib/request.js

```diff
@@ -88,12 +88,19 @@ export default function (relativeUrl, opts, callback) {
 
     return fetch(url, config)
         .then((response) => {
-            if (response.headers.get('content-type').indexOf('application/json') !== -1) {
-                return response.json();
+            let result = null;
+
+            if (response.status >= 400 && response.status < 500 && response.headers.get('X-BC-Preview-Mode').indexOf('true') !== -1) {
```

Comment:
do you want to keep status codes range wide?

Reply:
yeah good call, I had this before I decided just on 401. I will make it more narrow.